modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -4026,95 +4026,162 @@ public void testGetExtensionDynamicObjectDefinitionTableValues()
 			_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
 				TestPropsValues.getCompanyId(), User.class.getName());
 
-		ObjectField objectField1 = _addCustomObjectField(
-			new LongIntegerObjectFieldBuilder(
-			).labelMap(
-				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
-			).name(
-				"longField"
-			).objectDefinitionId(
-				objectDefinition.getObjectDefinitionId()
-			).build());
-		ObjectField objectField2 = _addCustomObjectField(
-			new TextObjectFieldBuilder(
-			).labelMap(
-				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
-			).name(
-				"textField"
-			).objectDefinitionId(
-				objectDefinition.getObjectDefinitionId()
-			).required(
-				true
-			).build());
-
-		User user = UserTestUtil.addUser();
+		boolean originalEnableLocalization =
+			objectDefinition.isEnableLocalization();
 
 		try {
+			objectDefinition.setEnableLocalization(true);
+
+			objectDefinition =
+				_objectDefinitionLocalService.updateObjectDefinition(
+					objectDefinition);
+
+			ObjectField objectField1 = _addCustomObjectField(
+				new TextObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).localized(
+					true
+				).name(
+					"localizedTextField"
+				).objectDefinitionId(
+					objectDefinition.getObjectDefinitionId()
+				).build());
+			ObjectField objectField2 = _addCustomObjectField(
+				new LongIntegerObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).name(
+					"longField"
+				).objectDefinitionId(
+					objectDefinition.getObjectDefinitionId()
+				).build());
+			ObjectField objectField3 = _addCustomObjectField(
+				new TextObjectFieldBuilder(
+				).labelMap(
+					LocalizedMapUtil.getLocalizedMap(
+						RandomTestUtil.randomString())
+				).name(
+					"textField"
+				).objectDefinitionId(
+					objectDefinition.getObjectDefinitionId()
+				).required(
+					true
+				).build());
+
+			User user = UserTestUtil.addUser();
+
+			try {
+				_objectEntryLocalService.
+					addOrUpdateExtensionDynamicObjectDefinitionTableValues(
+						TestPropsValues.getUserId(), objectDefinition,
+						user.getUserId(), Collections.emptyMap(),
+						ServiceContextTestUtil.getServiceContext());
+
+				Assert.fail();
+			}
+			catch (ObjectEntryValuesException.Required
+						objectEntryValuesException) {
+
+				Assert.assertEquals(
+					"No value was provided for required object field " +
+						"\"textField\"",
+					objectEntryValuesException.getMessage());
+			}
```

Comment:
@pedro-oliveira446 can we change to use the `AssertUtils` here?

```java

ObjectDefinition finalObjectDefinition = objectDefinition;
User user = UserTestUtil.addUser();

AssertUtils.assertFailure(
	ObjectEntryValuesException.Required.class,
	"No value was provided for required object field \"textField\"",
	() ->
		_objectEntryLocalService.
			addOrUpdateExtensionDynamicObjectDefinitionTableValues(
				TestPropsValues.getUserId(), finalObjectDefinition,
				user.getUserId(), Collections.emptyMap(),
				ServiceContextTestUtil.getServiceContext())
);
```

Reply:
Just done!